src/fixed.h

```diff
@@ -149,34 +149,39 @@ class fixedf {
 	}
 	friend fixedf operator/(const fixedf &a, const fixedf &b)
 	{
-		// 128-bit divided by 64-bit, to make sure high bits are not lost
-		Sint64 quotient_hi = a.v >> (64 - FRAC);
-		Uint64 quotient_lo = a.v << FRAC;
-		Sint64 d = b.v;
-		int isneg = 0;
-		Sint64 remainder = 0;
-
-		if (d < 0) {
-			d = -d;
-			isneg = 1;
+		Uint64 abs_a = a.v, abs_b = b.v;
+		bool is_neg = false;
+		if (a.v < 0) {
+			abs_a = -abs_a;
+			is_neg = !is_neg;
 		}
-
+		if (b.v < 0) {
+			abs_b = -abs_b;
+			is_neg = !is_neg;
+		}
```

Comment:
I would strongly recommend using bitwise ops and std::abs rather than explicit branches, as it will generally compile into optimized code under more scenarios. E.g. something like:
```c++
uint64_t abs_a = std::abs(a.v), abs_b = std::abs(b.v);
bool is_neg = int(a.v < 0) ^ int(b.v < 0);
```

(Note: this is well-defined, `true` is defined to promote to the integral value `1`.)
See this Godbolt for example of how this affects code that would be generated in debug mode: https://godbolt.org/z/6MsTPv8dz

Yes, under `-O3` or equivalent this code would produce the same generated assembly, but the above suggestion will both produce better performance in debug mode (wasting less programmer time) and more clearly expresses what is logically occurring in the if statements without the extra cognitive overhead of the order-dependent logic.

Reply:
> I would strongly recommend using bitwise ops and std::abs rather than explicit branches, as it will generally compile into optimized code under more scenarios. E.g. something like:
> 
> ```c++
> uint64_t abs_a = std::abs(a.v), abs_b = std::abs(b.v);
> bool is_neg = int(a.v < 0) ^ int(b.v < 0);
> ```
> 
> (Note: this is well-defined, `true` is defined to promote to the integral value `1`.) See this Godbolt for example of how this affects code that would be generated in debug mode: https://godbolt.org/z/6MsTPv8dz
> 
> Yes, under `-O3` or equivalent this code would produce the same generated assembly, but the above suggestion will both produce better performance in debug mode (wasting less programmer time) and more clearly expresses what is logically occurring in the if statements without the extra cognitive overhead of the order-dependent logic.

Done. Please note that with the requested changes dividing fixed(INT64_MIN) by anything or dividing anything by fixed(INT64_MIN) causes undefined behaviour because std::abs(INT64_MIN) is undefined behaviour.